torchao/_models/llama/generate.py

```diff
@@ -244,7 +244,7 @@ def encode_tokens(tokenizer, string, bos=True, device=default_device):
 
 def _load_model(checkpoint_path, device, precision):
     checkpoint = torch.load(
-        str(checkpoint_path), mmap=True, weights_only=True, map_location=device
+        str(checkpoint_path), mmap=True, weights_only=True, map_location="cpu"
```

Comment:
actually is the map_location change correct?

Reply:
oh OK, makes sense, this should be OK, some quantized format does not support loading in CPU but for full precision model it should be fine